### PR TITLE
Fixes for SPI class

### PR DESCRIPTION
**Pull request by caternuson**
Fix for #10 . But also fixed to work with actual phase/polarity of sensor. Example also updated to show SPI usage.

### CODE REVIEW

**Comment by ladyada** [comment]
@kattni please test this if hasnt yet, and then merge/release!

**Comment by kattni** [comment]
Will do! Still needs testing. Added to my list.

File: adafruit_lsm9ds0.py [hunk 1]
```diff
@@ -397,8 +397,8 @@ class LSM9DS0_SPI(LSM9DS0):
     """Driver for the LSM9DS0 connected over SPI."""
     # pylint: disable=no-member
     def __init__(self, spi, xmcs, gcs):
-        self._gyro_device = spi_device.I2CDevice(spi, gcs)
-        self._xm_device = spi_device.I2CDevice(spi, xmcs)
+        self._mag_device = spi_device.SPIDevice(spi, xmcs, baudrate=200000, phase=1, polarity=1)
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: The rest of the code refers to `self._gyro_device`, is this incorrect or is the rest of the code incorrect?

File: adafruit_lsm9ds0.py [hunk 2]
```diff
@@ -397,8 +397,8 @@ class LSM9DS0_SPI(LSM9DS0):
...
+        self._xg_device = spi_device.SPIDevice(spi, gcs, baudrate=200000, phase=1, polarity=1)
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: The rest of the code refers to `self._xm_device`. Same question as above.

**Review by kattni — CHANGES_REQUESTED** [review]
Thanks for doing this!

If I update both of the following two suggestions to match the rest of the code, it runs perfectly. So please take a look and update either the two lines I suggested, or the rest of the SPI code to match.

File: adafruit_lsm9ds0.py (hunk 1 above)
Comment (caternuson) [reply adafruit_lsm9ds0.py]: see comment below

File: adafruit_lsm9ds0.py (hunk 2 above)
Comment (caternuson) [reply adafruit_lsm9ds0.py]: see comment below

**Comment by caternuson** [comment]
Thanks @kattni . Silly copy-pasta error. Didn't have hardware to test so only did like a 2 second copy paste of the fix from LSM9DS1. For some reason, that library uses different variable names for the gyro and mag sensors.

Feel stupid for not even doing a visual scan - that should've stuck out like a major thumb of soreness.

**Comment by kattni** [comment]
So the fix resulted in the code failing without swapping the two digital pins. When it runs, all of the data is incorrect. Example:
```
Acceleration (m/s^2): (0.154,0.154,0.154)
Magnetometer (gauss): (-0.185,-0.185,-0.185)
Gyroscope (degrees/sec): (-20.247,-20.247,-20.247)
Temperature: -39.250C
```
As @caternuson doesn't have hardware to test it with, he's asked that I take over this fix. I'll take a look at it when I finish my current project.

**Comment by caternuson** [comment]
Thanks @kattni . Ping me if you need help. But yah, doesn't look like a simple copy-paste is all that's needed here. :(

File: examples/lsm9ds0_simpletest.py [hunk 3]
```diff
@@ -12,6 +12,17 @@
...
+# from digitalio import DigitalInOut, Direction
+# spi = busio.SPI(board.SCK, board.MOSI, board.MISO)
+# csag = DigitalInOut(board.D5)
+# csag.direction = Direction.OUTPUT
+# csag.value = True
+# csm = DigitalInOut(board.D6)
+# csm.direction = Direction.OUTPUT
+# csm.value = True
+# sensor = adafruit_lsm9ds0.LSM9DS0_SPI(spi, csag, csm)
+
```
Comment (caternuson) [inline examples/lsm9ds0_simpletest.py:21]: This example copy-pasted from LSM9DS1 is probably all wrong also. For one thing, the CS pins are taken in a different order for the LSM9DS0:
```python
def __init__(self, spi, xmcs, gcs):
```

**Comment by ladyada** [comment]
here's code, there were a few things wrong!

[lsm9.zip](https://github.com/adafruit/Adafruit_CircuitPython_LSM9DS0/files/2609452/lsm9.zip)

**Comment by caternuson** [comment]
@ladyada thanks!

@kattni I simply added code from .zip as is and then fixed a few lint things.

File: adafruit_lsm9ds0.py [hunk 4]
```diff
@@ -379,8 +372,9 @@ def _read_bytes(self, sensor_type, address, count, buf):
             device = self._xm_device
         with device as i2c:
             buf[0] = address & 0xFF
-            i2c.write(buf, end=1, stop=False)
+            i2c.write(buf, end=1)
             i2c.readinto(buf, end=count)
+        #print("read from %02x: %s" % (address, [hex(i) for i in buf[:count]]))
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: Please remove `print` statement. Seems like it's left over from debugging, shouldn't be needed in the final release.

File: adafruit_lsm9ds0.py [hunk 5]
```diff
@@ -391,26 +385,23 @@ def _write_u8(self, sensor_type, address, val):
             self._BUFFER[0] = address & 0xFF
             self._BUFFER[1] = val & 0xFF
             i2c.write(self._BUFFER, end=2)
+        #print("write to %02x: %02x" % (address, val))
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: Same as above.

File: adafruit_lsm9ds0.py [hunk 6]
```diff
@@ -419,18 +410,18 @@ def _read_bytes(self, sensor_type, address, count, buf):
         else:
             device = self._xm_device
         with device as spi:
-            spi.configure(baudrate=200000, phase=0, polarity=0)
-            buf[0] = (address | 0x80) & 0xFF
+            buf[0] = (address | 0x80 | 0x40) & 0xFF
             spi.write(buf, end=1)
             spi.readinto(buf, end=count)
+        #print("read from %02x: %s" % (address, [hex(i) for i in buf[:count]]))
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: Same as above.

File: adafruit_lsm9ds0.py [hunk 7]
```diff
@@ -419,18 +410,18 @@ def _read_bytes(self, sensor_type, address, count, buf):
...
         if sensor_type == _GYROTYPE:
             device = self._gyro_device
         else:
             device = self._xm_device
         with device as spi:
-            spi.configure(baudrate=200000, phase=0, polarity=0)
             self._BUFFER[0] = (address & 0x7F) & 0xFF
             self._BUFFER[1] = val & 0xFF
             spi.write(self._BUFFER, end=2)
+        #print("write to %02x: %02x" % (address, val))
```
Comment (kattni) [inline adafruit_lsm9ds0.py]: Same as above.

**Review by kattni — CHANGES_REQUESTED** [review]
Tested this on Feather M4, works great! I have a couple of suggestions below.

**Comment by kattni** [comment]
@ladyada Do you want the `print` statements left in?

**Comment by ladyada** [comment]
keep them in, they're not taking up any space :)

**Review by kattni — APPROVED** [review]

**Pull request merged**